env/env_test.cc

```diff
@@ -1046,20 +1049,21 @@ class TestLogger : public Logger {
     }
 
     for (size_t i = 0; i < sizeof(new_format); i++) {
-      if (new_format[i] == 'x') {
+      if (new_format[i] == marker_) {
```

Comment:
no need to add a new constructor, looks like this class is only used for this test, we can simply change the 'x' to '#'